### DISCUSSION TITLE
Use modules/npm?

### DISCUSSION THREAD

**Discussion by scriptsengineer** [quentincaffeino/godot-console#112] [Ideas]
The issue #111 discussed a possible solution of using npm to install addons on godot, I would like to know more about this @quentincaffeino.

**Comment by quentincaffeino** [comment]
This is possible thanks to the fact that you can change modules dir, see `.npmrc`.

Also you don't have to put your addon in root folder as with npm you can publish just a specific directory.

I started separating modules from the console beginning with callbacks. You can see the module in a separate directory already in `addons/@quentincaffeino/godot-callback. It is not yet published but that is not hard.

In case it would be moved to a separate repo there would be one problem, that that would not work with godot. For that separate build repo would be needed that with magic of cicd pulls all dependencies together for godot to be able to just grab a zip of repo.

Next nice thing is that with power of monorepos multiple packages can live in a same repo.
